Kinvey/Kinvey/RealmCache.swift

```diff
@@ -66,16 +68,41 @@ internal class RealmCache<T: Persistable>: Cache<T> where T: NSObject {
         return realmResults
     }
     
-    fileprivate func newInstance<P: Persistable>(_ type: P.Type) -> P {
+    fileprivate func newInstance<P:Persistable>(_ type: P.Type) -> P {
         return type.init()
     }
-    
-    override func detach(_ entity: T) -> T {
+
+    fileprivate func detach(_ entity: Object, props: [String]) -> Object {
```

Comment:
Good point. `detach(entity, query)` is used in other parts of the SDK. I'd like to make it private as well, since `detach` shouldn't be an API on the `Cache` anyway. However, it'll require a better understanding of the references.

As for deprecation, I don't think it's necessary, because the `Cache` API itself is not public.